### ISSUE TITLE
Fail to build with nim 2.0.0

### ISSUE THREAD

**Issue by ncihnegn** [treeform/puppy#105]
```
puppy/src/puppy/platforms/macos/objc.nim(123, 32) Error: the special identifier '_' is ignored in declarations and cannot be used
```

**Comment by fukusuket** [comment]
I am also running into the same issue(macOS Ventura)
```
%nimble build
  Verifying dependencies for sample@0.1.0
     Info:  Dependency on puppy@any version already satisfied
  Verifying dependencies for puppy@2.0.3
     Info:  Dependency on libcurl@>= 1.0.0 already satisfied
  Verifying dependencies for libcurl@1.0.0
     Info:  Dependency on zippy@>= 0.10.0 already satisfied
  Verifying dependencies for zippy@0.10.10
     Info:  Dependency on webby@>= 0.1.6 already satisfied
  Verifying dependencies for webby@0.1.9
   Building sample/sample using c backend
/Users/fukusuke/.nimble/pkgs2/puppy-2.0.3-988d9d6798ea228aa2c6e13a53021ebb2a0c3ca6/puppy/platforms/macos/macdefs.nim(28, 1) template/generic instantiation of `objc` from here
/Users/fukusuke/.nimble/pkgs2/puppy-2.0.3-988d9d6798ea228aa2c6e13a53021ebb2a0c3ca6/puppy/platforms/macos/objc.nim(123, 32) Error: the special identifier '_' is ignored in declarations and cannot be used
       Tip: 10 messages have been suppressed, use --verbose to show them.
nimble.nim(229)          buildFromDir

    Error:  Build failed for the package: sample
% nim -v
Nim Compiler Version 2.0.0 [MacOSX: amd64]
Compiled at 2023-08-19
Copyright (c) 2006-2023 by Andreas Rumpf

active boot switches: -d:release
% nimble -v
nimble v0.14.2 compiled at 2023-08-19 16:19:52
git hash: couldn't determine git hash
```

**Comment by guzba** [comment]
This fix is now tagged in 2.1.0 release
